lib/widgets/action_sheet.dart

```diff
@@ -310,6 +313,37 @@ class CopyMessageTextButton extends MessageActionSheetMenuItemButton {
   }
 }
 
+class CopyMessageLinkButton extends MessageActionSheetMenuItemButton {
+  CopyMessageLinkButton({
+    super.key,
+    required super.message,
+    required super.messageListContext,
+  });
+
+  @override IconData get icon => Icons.link;
+
+  @override
+  String label(ZulipLocalizations zulipLocalizations) {
+    return zulipLocalizations.actionSheetOptionCopyMessageLink;
+  }
+
+  @override void onPressed(BuildContext context) {
+    Navigator.of(context).pop();
+    final zulipLocalizations = ZulipLocalizations.of(messageListContext);
+
+    final store = PerAccountStoreWidget.of(messageListContext);
+    final messageLink = narrowLink(
+      store,
+      SendableNarrow.ofMessage(message, selfUserId: store.selfUserId),
+      nearMessageId: message.id,
+    );
+
+    copyWithPopup(context: messageListContext,
+        successContent: Text(zulipLocalizations.successLinkCopied),
```

Comment:
Let's make a new string in `zulipLocalizations`, probably with the name `successMessageLinkCopied`, saying "Message link copied", similar to the "Message text copied" from my previous comment. The `successLinkCopied` string is meant for a different interaction (long-pressing a link that appears in rendered message content).